### PR TITLE
cmake: when not SC_QT skip install GUI help but not Guides

### PR DESCRIPTION
**Pull request by sonoro1234**
## Purpose and Motivation

When SC_QT is off cmake regex is skipping GUI in filename so that Guides are also skipped (issue #5043)

## Types of changes

Change cmake regex to skip GUI followed by . or _ or - which are the patterns I could find in gui help files

### CODE REVIEW

**Comment by brianlheim** [comment]
IIUC, the original intention of this regex was to exclude specific **directories** of **SCClassLibrary**. my understanding seems to be supported by this history:

in https://github.com/supercollider/supercollider/commit/1e5135dac , which introduced this exclusion feature, it is clearly intended to exclude directories (.git and QtCollider).

https://github.com/supercollider/supercollider/commit/061c5b50cc2 clearly treats this regex as applying to only directories (in the symlinking code)

https://github.com/supercollider/supercollider/pull/3456, which switched the regex from "QtCollider" to "GUI", also seemed to assume that this would only ever apply to folders.

using this regex when installing from HelpSource appears to be a mistake. it was originally done in this commit: https://github.com/supercollider/supercollider/commit/d7b078c99

which appears to be an accidental copy-paste of the SCClassLibrary installation command. i say it is accidental because at that point in time the regex could only ever contain "QtCollider", but there was no folder or file name QtCollider in HelpSource at that time. probably it went unnoticed because of that.

there is also another aspect here which is that this regex appears to behave case-insensitively on your system (but not on mine). we should definitely support case insensitive filesystems by not behaving differently (i.e. installing or not installing different files and directories).

i believe the correct fix here is:
- do not apply this regex to HelpSource (that fixes the problem you were intending to solve)
- change the regexes to the more specific `/GUI$` and `/Ableton$` so that this indeed only applies to directories as everyone already appears to think it does
- remove the regex check on line 484 of CMakeLists.txt, because it currently does nothing and only causes confusion (this will check only the immediate subdirectories of SCClassLibrary, none of which are named GUI or Ableton)
- deal with one outlier in `SCClassLibrary`, `UnitTestGUI.sc`, by creating `SCClassLibrary/Common/UnitTesting/GUI` and moving `UnitTestGUI.sc` into it.
- consider doing the same for `SCClassLibrary/Platform/linux/LIDGui.sc` and `SCClassLibrary/Common/Quarks/QuarksGui.sc`, as on a case-insensitive filesystem these are currently being excluded in a non-GUI install, and they are definitely GUI- and GUI-only code.

**Comment by sonoro1234** [comment]
@brianlheim You have done a throughout and deep investigation on this issue.
These are a lot of changes!!!. Do you want me to do all of them or perhaps I should close this PR and let you implement them?

**Comment by brianlheim** [comment]
> These are a lot of changes!!!. Do you want me to do all of them or perhaps I should close this PR and let you implement them?

please, i would prefer that you implement these changes. it's about 5 lines of code and moving 1 or 3 files, and you can easily test the result if you have a command like `diff` on your system. i tried to be as clear as possible in my description so that you could do it in this PR. let me know if you need any more details.

> May be another solution could be just dont change anything when SC_QT is off?
> Would it be harmful to have help and Classes related to GUI even if SC_QT is off?

help -- GUI related help files are already installed, and that's fine. there is not an easy way to exclude those files anyway at the moment. right now the effect of using this regex is to prevent installation of a more or less arbitrary subset of help files that have "GUI" in the name.

classes -- GUI class files **absolutely** cannot be installed, because any file installed that has an `initClass` which references GUI primitives will break class library compilation. the easiest solution, which we have already done in almost all cases, is to just separate out anything which uses GUI classes and not install it.

**Comment by sonoro1234** [comment]
Done!! (I think so)

File: CMakeLists.txt [hunk 1]
```diff
@@ -481,10 +481,8 @@ if(APPLE)
 		#symlink the folders inside SCCLassLibrary. Check if QtCollider folder should be symlinked or not.
 		file(GLOB classlibraryfolders RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}/SCClassLibrary" "SCClassLibrary/[^.]*")
 		foreach(onedirectory ${classlibraryfolders})
-			if(NOT "${onedirectory}" MATCHES "${SCCLASSLIB_EXCLUDE_REGEX}")
 				install( CODE
 					"EXECUTE_PROCESS(COMMAND ln -shF ${CMAKE_CURRENT_SOURCE_DIR}/SCClassLibrary/${onedirectory} ${CMAKE_INSTALL_PREFIX}/${auxresourcesdir}/SCClassLibrary/${onedirectory} )" )
```
Comment (brianlheim) [inline CMakeLists.txt:485]: can you please deindent these lines?

Comment (sonoro1234) [reply CMakeLists.txt]: Yes, done.

**Pull request merged**